pyap_beauhurst/source_CA/data.py

```diff
@@ -106,357 +107,375 @@
+    [Rr][Aa][Mm][Pp]{div}|
+    [Rr][Aa][Nn][Gg][Ee]{div}|[Rr][Gg]{div}|
+    [Rr][Aa][Nn][Gg]{div}|
+    [Rr][Ii][Dd][Gg][Ee]{div}|
+    [Rr][Ii][Ss][Ee]{div}|
+    [Rr][Oo][Aa][Dd]{div}|[Rr][Dd]{div}|
+    [Rr][Oo][Nn][Dd]\-[Pp][Oo][Ii][Nn][Tt]{div}|[Rr][Dd][Pp][Tt]{div}|
+    [Rr][Oo][Uu][Tt][Ee]{div}|[Rr][Tt][Ee]{div}|
+    [Rr][Oo][Ww]{div}|
+    [Rr][Uu][Ee][Ll][Ll][Ee]{div}|[Rr][Ll][Ee]{div}|
+    [Rr][Uu][Ee]{div}|
+    [Rr][Uu][Nn]{div}|
+    [Ss][Ee][Nn][Tt][Ii][Ee][Rr]{div}|[Ss][Ee][Nn][Tt]{div}|
+    # Street
+    [Ss][Tt][Rr][Ee][Ee][Tt]{div}|[Ss][Tt](?![A-Za-z]){div}|
+    # Square
+    [Ss][Qq][Uu][Aa][Rr][Ee]{div}|[Ss][Qq]{div}|
+    [Ss][Uu][Bb][Dd][Ii][Vv][Ii][Ss][Ii][Oo][Nn]{div}|[Ss][Uu][Bb][Dd][Ii][Vv]{div}|
+    [Tt][Ee][Rr][Rr][Aa][Cc][Ee]{div}|[Tt][Ee][Re][Re]{div}|
+    [Tt][Ee][Rr][Rr][Aa][Ss][Ss][Ee]{div}|[Tt][Ss][Ss][Es]{div}|
+    [Tt][Hh][Ii][Cc][Kk][Ee][Tt]{div}|[Tt][Hh][Ii][Cc][Kk]{div}|
+    [Tt][Oo][Ww][Ee][Rr][Ss]{div}|
+    [Tt][Oo][Ww][Nn][Ll][Ii][Nn][Ee]{div}|[Tt][Ll][Ii][Nn][Ee]{div}|
+    [Tt][Rr][Aa][Ii][Ll]{div}|
+    [Tt][Uu][Rr][Nn][Aa][Bb][Oo][Uu][Tt]{div}|[Tt][Rr][Nn][Aa][Bb][Tt]{div}|
+    [Vv][Aa][Ll][Ee]{div}|
+    [Vv][Ii][Aa]{div}|
+    [Vv][Ii][Ee][Ww]{div}|
+    [Vv][Ii][Ll][Ll][Aa][Gg][Ee]{div}|[Vv][Ii][Ll][Ll][Gg][Ee]{div}|
+    [Vv][Ii][Ll][Ll][Aa][Ss]{div}|
+    [Vv][Ii][Ss][Tt][Aa]{div}|
+    [Vv][Oo][Ii][Ee]{div}|
+    [Ww][Aa][Ll][Lk]{div}|
+    [Ww][Aa][Yy]{div}|
+    [Ww][Hh][Aa][Rr][Ff]{div}|
+    [Ww][Oo][Oo][Dd]{div}|
+    [Ww][Yy][Nn][Dd]{div}
+)
+(?P<route_id>
+    [\(\s\,]{route_symbols}
+    [Rr][Oo][Uu][Tt][Ee]\s[A-Za-z0-9]+[\)\s\,]{route_symbols}
+)?
+""".format(
+    div=r"[\.\s,]{0,2}", route_symbols="{0,3}"
 )
 
 floor = r"""
-            (?P<floor>
-                (?:
-                \d+[A-Za-z]{0,2}\.?\ [Ff][Ll][Oo][Oo][Rr]\ 
-                )
-                |
-                (?:
-                    [Ff][Ll][Oo][Oo][Rr]\ \d+[A-Za-z]{0,2}\ 
-                )
-            )
-        """
+(?P<floor>
+    (?:
+    \d+[A-Za-z]{0,2}\.?\s[Ff][Ll][Oo][Oo][Rr]\s
+    )
+    |
+    (?:
+        [Ff][Ll][Oo][Oo][Rr]\s\d+[A-Za-z]{0,2}\s
+    )
+)
+"""
 
 building = r"""
-            (?:
-                (?:
-                    (?:[Bb][Uu][Ii][Ll][Dd][Ii][Nn][Gg])
-                    |
-                    (?:[Bb][Ll][Dd][Gg])
-                )
-                \ \d{0,2}[A-Za-z]?
-            )
-            """
+(?P<building_id>
+    (?:
+        (?:[Bb][Uu][Ii][Ll][Dd][Ii][Nn][Gg])
+        |
+        (?:[Bb][Ll][Dd][Gg])
+    )
+    \s
+    (?:
+        (?:
+            [Aa][Nn][Dd]\s
+            |
+            {thousand}
+            |
+            {hundred}
+            |
+            {zero_to_nine}
+            |
+            {ten_to_ninety}
+        ){{1,5}}
+        |
+        \d{{0,4}}[A-Za-z]?
+    )
+    \s ?
+)  # end building_id
+""".format(
+    thousand=thousand,
+    hundred=hundred,
+    zero_to_nine=zero_to_nine,
+    ten_to_ninety=ten_to_ninety,
```

Comment:
happy to replace them if you think its a good idea, but I avoided it just because I think reading regex is complicated enough without also adding in `{more brackets}`.

Reply:
especially in the case where the formatting is adding in parts of regex, like `from_to="{1,5}"`

Reply:
not for this PR, but it would be good to see if there is a way we can differentiate between regex curly brackts and the formattin gones. I thought double brackets are for that?

Reply:
yeah, that's what double brackets are supposed to be for. With a bit of a refactor I think it could be possible. I'll make an issue an look at it when I get a minute

Reply:
eugh can't make an issue on the fork. But I will look at doing it when i get a minute

Reply:
the repo didnt' have issues active - i've activated it. You should be able to now

Reply:
:+1: #4